doc/source/tune-usage.rst

```diff
@@ -296,6 +296,40 @@ of a trial, you can additionally set the checkpoint_at_end to True. An example i
         },
     })
 
+Recovering From Failures (Experimental)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A Tune experiment run may crash due to unforeseen circumstances. To avoid losing all progress, set ``checkpoint_mode=True`` in ``run_experiments``. Note that only **checkpointable experiments** (i.e., experiments with ``checkpoint_freq > 0``) will be preserved. With ``checkpoint_mode=True``, Tune frequently save the entire experiment state at the ``local_dir`` of the first checkpointable experiment provided.  E.g.:
+
+.. code-block:: python
+
+    run_experiments({
+        "my_experiment_name": {
+            "run": my_trainable
+            "checkpoint_freq": 10,
+            "local_dir": "~/path/to/results"
+        },
+    }, checkpoint_mode=True)
```

Comment:
This is confusing. Just always checkpoint, and set resume=True. The cost should be minimal.

Alternatively, make resume=True the same as checkpoint_mode=True.

Reply:
Also, it makes it so that you can't have resume=True always. Users don't want to have to deal with flags every time they run a tune file, they just want it things to work, and if it fails, to be resumable.

Reply:
Done

Reply:
I would prefer that this is opt-in for at least a week or two when on master. After that, `resume` can be default.